<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Downstream analysis | An RNA-seq Data Analysis Protocol</title>
  <meta name="description" content="This protocol provides an overview of the steps involved in running an analysis of Bulk RNA-seq short reads from raw-data to counts" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Downstream analysis | An RNA-seq Data Analysis Protocol" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This protocol provides an overview of the steps involved in running an analysis of Bulk RNA-seq short reads from raw-data to counts" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Downstream analysis | An RNA-seq Data Analysis Protocol" />
  
  <meta name="twitter:description" content="This protocol provides an overview of the steps involved in running an analysis of Bulk RNA-seq short reads from raw-data to counts" />
  

<meta name="author" content="David Chisanga and Wei Shi" />


<meta name="date" content="2021-02-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="more-details-on-the-protocol.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bulk RNA-seq analysis protocol</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a><ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>2.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>3</b> Running Environment</a><ul>
<li class="chapter" data-level="3.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>3.1</b> RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="running-environment.html"><a href="running-environment.html#unix-server-laptop"><i class="fa fa-check"></i><b>3.2</b> UNIX server + laptop</a></li>
<li class="chapter" data-level="3.3" data-path="running-environment.html"><a href="running-environment.html#laptop-only"><i class="fa fa-check"></i><b>3.3</b> Laptop only</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="rna-seq-analysis-protocol.html"><a href="rna-seq-analysis-protocol.html"><i class="fa fa-check"></i><b>4</b> RNA-SEQ analysis protocol</a></li>
<li class="chapter" data-level="5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html"><i class="fa fa-check"></i><b>5</b> More details on the protocol</a><ul>
<li class="chapter" data-level="5.1" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sequencing-quality"><i class="fa fa-check"></i><b>5.1</b> Sequencing quality</a></li>
<li class="chapter" data-level="5.2" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-annotation"><i class="fa fa-check"></i><b>5.2</b> Gene annotation</a></li>
<li class="chapter" data-level="5.3" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#index-building"><i class="fa fa-check"></i><b>5.3</b> Index building</a></li>
<li class="chapter" data-level="5.4" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-mapping"><i class="fa fa-check"></i><b>5.4</b> Read mapping</a></li>
<li class="chapter" data-level="5.5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-counting"><i class="fa fa-check"></i><b>5.5</b> Read counting</a></li>
<li class="chapter" data-level="5.6" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>5.6</b> Gene filtering</a></li>
<li class="chapter" data-level="5.7" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sample-clustering"><i class="fa fa-check"></i><b>5.7</b> Sample clustering</a></li>
<li class="chapter" data-level="5.8" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#statistical-testing-of-differential-expression"><i class="fa fa-check"></i><b>5.8</b> Statistical testing of differential expression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="downstream-analysis.html"><a href="downstream-analysis.html"><i class="fa fa-check"></i><b>6</b> Downstream analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="downstream-analysis.html"><a href="downstream-analysis.html#load-previous-results"><i class="fa fa-check"></i><b>6.1</b> Load previous results</a></li>
<li class="chapter" data-level="6.2" data-path="downstream-analysis.html"><a href="downstream-analysis.html#visualisations"><i class="fa fa-check"></i><b>6.2</b> Visualisations</a><ul>
<li class="chapter" data-level="6.2.1" data-path="downstream-analysis.html"><a href="downstream-analysis.html#heatmaps"><i class="fa fa-check"></i><b>6.2.1</b> Heatmaps</a></li>
<li class="chapter" data-level="6.2.2" data-path="downstream-analysis.html"><a href="downstream-analysis.html#mean-difference-plot"><i class="fa fa-check"></i><b>6.2.2</b> Mean-difference plot</a></li>
<li class="chapter" data-level="6.2.3" data-path="downstream-analysis.html"><a href="downstream-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>6.2.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="downstream-analysis.html"><a href="downstream-analysis.html#pathway-analysis"><i class="fa fa-check"></i><b>6.3</b> Pathway Analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="downstream-analysis.html"><a href="downstream-analysis.html#gene-ontology-enrichment"><i class="fa fa-check"></i><b>6.3.1</b> Gene Ontology enrichment</a></li>
<li class="chapter" data-level="6.3.2" data-path="downstream-analysis.html"><a href="downstream-analysis.html#kegg-pathway-enrichment"><i class="fa fa-check"></i><b>6.3.2</b> KEGG pathway enrichment</a></li>
<li class="chapter" data-level="6.3.3" data-path="downstream-analysis.html"><a href="downstream-analysis.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.3.3</b> Gene Set Enrichment Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An RNA-seq Data Analysis Protocol</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downstream-analysis" class="section level1">
<h1><span class="header-section-number">6</span> Downstream analysis</h1>
<p>In this chapter we will run through a series of example downstream analyses that can be performed on the identified differentially expressed genes.</p>
<div id="load-previous-results" class="section level2">
<h2><span class="header-section-number">6.1</span> Load previous results</h2>
<p>The results from the differential analysis were saved as R-objects. These are imported below and used for the rest of the analysis</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="co">#load the required package</span></a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb33-3" data-line-number="3"><span class="kw">load</span>(<span class="st">&quot;Counts2.RData&quot;</span>) <span class="co"># loads the counts dataset</span></a></code></pre></div>
</div>
<div id="visualisations" class="section level2">
<h2><span class="header-section-number">6.2</span> Visualisations</h2>
<p>There are a number of ways in which you can visualise the results from the differential expression analysis, while in this chapter we attempt to cover the most common ways of visualising the data, they are by no means exhaustive.</p>
<div id="heatmaps" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Heatmaps</h3>
<p>Heatmaps are a great way of demonstrating the differences in the expression patterns between different conditions in your dataset. Here we use the <em>coolmap</em> function within the <strong>limma</strong> package which is an extension of the <em>heatmap.2</em> function within the <strong>gplots</strong> package. The heatmap below shows the expression pattern of the top 100 differentially expressed genes between CIS vs WT.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">heat.data &lt;-<span class="st"> </span>y<span class="op">$</span>E[<span class="kw">head</span>(<span class="kw">rownames</span>(de.genes), <span class="dt">n =</span> <span class="dv">100</span>), ]</a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw">colnames</span>(heat.data) &lt;-<span class="st"> </span>targets<span class="op">$</span>Sample</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="kw">rownames</span>(heat.data) &lt;-<span class="st"> </span><span class="kw">head</span>(de.genes<span class="op">$</span>Symbol, <span class="dt">n =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb34-4" data-line-number="4"><span class="kw">coolmap</span>(</a>
<a class="sourceLine" id="cb34-5" data-line-number="5">  heat.data,</a>
<a class="sourceLine" id="cb34-6" data-line-number="6">  <span class="dt">margin =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">4</span>),</a>
<a class="sourceLine" id="cb34-7" data-line-number="7">  <span class="dt">cexCol =</span> <span class="fl">0.8</span>,</a>
<a class="sourceLine" id="cb34-8" data-line-number="8">  <span class="dt">lhei =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">4</span>),</a>
<a class="sourceLine" id="cb34-9" data-line-number="9">  <span class="dt">cexRow =</span> <span class="fl">0.6</span></a>
<a class="sourceLine" id="cb34-10" data-line-number="10">)</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:figheatmap"></span>
<img src="bookdown-RNA-seq-protocol_files/figure-html/figheatmap-1.png" alt="Heatmap plot of differentially expressed genes in CIS vs WT" width="576" />
<p class="caption">
Figure 6.1: Heatmap plot of differentially expressed genes in CIS vs WT
</p>
</div>
</div>
<div id="mean-difference-plot" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Mean-difference plot</h3>
<p>A mean-difference plot or MD-plot is a plot that can be used to show the fold-change differences against the average expression values of all genes used in the analysis. This can be generated easily using the <em>plotMD</em> function within the <strong>limma</strong> package.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw">plotMD</span>(fit.contr, <span class="dt">column =</span> <span class="dv">1</span>,<span class="dt">status =</span> dt,<span class="dt">cex=</span><span class="fl">0.8</span>)</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:figmd"></span>
<img src="bookdown-RNA-seq-protocol_files/figure-html/figmd-1.png" alt="Mean-Difference plot. Represents relationship between mean expression and fold-changes" width="576" />
<p class="caption">
Figure 6.2: Mean-Difference plot. Represents relationship between mean expression and fold-changes
</p>
</div>
<p>Alternatively, an interactive mean-difference plot can be generated using <em>glMDPlot</em> in <strong>Glimma</strong> which outputs an interactive html page with results displayed in the left panel and the expression values for the selected gene. A gene within the dataset can be searched through a search bar.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw">library</span>(Glimma)</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"><span class="kw">glMDPlot</span>(fit.contr, <span class="dt">coef=</span><span class="dv">1</span>, <span class="dt">status=</span>dt, <span class="dt">main=</span><span class="kw">colnames</span>(fit.contr)[<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb36-3" data-line-number="3">         <span class="dt">side.main=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="dt">counts=</span>y<span class="op">$</span>E, <span class="dt">groups=</span><span class="kw">colnames</span>(contr),</a>
<a class="sourceLine" id="cb36-4" data-line-number="4">         <span class="dt">launch =</span> F)</a></code></pre></div>
<p><a href="glimma-plots/MD-Plot.html">Click here for interactive version</a></p>
<p>Hover over points to see sample-wise expression</p>
<ul>
<li><p>Click on column names to sort by column</p></li>
<li><p>Click rows on tables to highlight gene</p></li>
</ul>
</div>
<div id="volcano-plot" class="section level3">
<h3><span class="header-section-number">6.2.3</span> Volcano plot</h3>
<p>A volcano plot shows the relationship between the log fold-change on the x-axis and a measure of statistical significance on the y-axis. The measure of significance can be -log(p-value) or the B-statistics. Here, we will use the <em>volcanoplot</em> function within the <strong>limma</strong> package.In the volcano plot below, differentially expressed genes are highlighted with red for up-regulated and blue for down-regulated genes.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co">#highlight de genes</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2">col&lt;-<span class="kw">c</span>(<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;red&quot;</span>)[<span class="kw">factor</span>(dt[,<span class="st">&quot;CISvsWT&quot;</span>])]</a>
<a class="sourceLine" id="cb37-3" data-line-number="3"><span class="kw">volcanoplot</span>(fit.contr,<span class="dt">col=</span>col)</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:figvolc"></span>
<img src="bookdown-RNA-seq-protocol_files/figure-html/figvolc-1.png" alt="Volcano plot. Shows the relationship between fold-change and significance of change" width="480" />
<p class="caption">
Figure 6.3: Volcano plot. Shows the relationship between fold-change and significance of change
</p>
</div>
</div>
</div>
<div id="pathway-analysis" class="section level2">
<h2><span class="header-section-number">6.3</span> Pathway Analysis</h2>
<p>From the differential expression analysis there were some 3181 differentially expressed genes. This is quiet a long list of genes to interpret and nearly impossible to go through a gene at a time to gain any meaningful biological understanding. A common downstream analysis step is to understand pathways/gene networks that the genes are implicated in.</p>
<div id="gene-ontology-enrichment" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Gene Ontology enrichment</h3>
<p>Gene ontology (<a href="http://www.geneontology.org/" class="uri">http://www.geneontology.org/</a>) provides a controlled vocabulary for describing biological processes (BP ontology), molecular functions (MF ontology) and cellular components (CC ontology).</p>
<p>The GO ontologies themselves are organism-independent; terms are associated with genes for a specific organism through direct experimentation or through sequence homology with another organism and its GO annotation.</p>
<p>Terms are related to other terms through parent-child relationships in a directed acylic graph.</p>
<p>You can use enrichment analysis as another way of drawing conclusions from your set of differentially expressed genes.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">enrich.pvalue&lt;-<span class="fl">0.00001</span></a></code></pre></div>
<p>Here, we use the <em>goana</em> function within the <strong>limma</strong> package to test for enrichment of differentially expressed genes between the CIS vs WT. Note that the <em>p-values</em> returned by <em>goana</em> are <strong>unadjusted for multiple testing</strong>. It is therfore, advisable that if the results are to be published, only terms with very small p-values should be included. For instance, in the example below, only terms with a p-value &lt; 10^{-5} are retained.</p>
<p>We use the output from the differential expression analysis and test for overrepresentation for the Up and Down regulated genes separately.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="cf">if</span>(<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;GO.db&quot;</span>))</a>
<a class="sourceLine" id="cb39-2" data-line-number="2">  BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;GO.db&quot;</span>)</a>
<a class="sourceLine" id="cb39-3" data-line-number="3"><span class="kw">library</span>(GO.db)</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">go.rst&lt;-<span class="kw">goana</span>(fit.contr,<span class="dt">species=</span><span class="st">&quot;Mm&quot;</span>)</a>
<a class="sourceLine" id="cb39-5" data-line-number="5"><span class="co">#exclude terms not meeting our cut-off</span></a>
<a class="sourceLine" id="cb39-6" data-line-number="6">go.rst&lt;-go.rst[<span class="kw">rowSums</span>(go.rst[,<span class="kw">c</span>(<span class="st">&quot;P.Up&quot;</span>,<span class="st">&quot;P.Down&quot;</span>)]<span class="op">&lt;</span>enrich.pvalue)<span class="op">==</span><span class="dv">1</span>,]</a>
<a class="sourceLine" id="cb39-7" data-line-number="7"><span class="kw">topGO</span>(go.rst)</a></code></pre></div>
<pre><code>##                                                              Term Ont    N  Up
## GO:0050794                         regulation of cellular process  BP 6084 593
## GO:0050789                       regulation of biological process  BP 6343 628
## GO:0065007                                  biological regulation  BP 6647 675
## GO:0048522                positive regulation of cellular process  BP 3439 335
## GO:0005515                                        protein binding  MF 5634 584
## GO:0048518              positive regulation of biological process  BP 3697 368
## GO:0060255          regulation of macromolecule metabolic process  BP 3917 361
## GO:0032501                       multicellular organismal process  BP 3570 335
## GO:0048856                       anatomical structure development  BP 3061 287
## GO:0019222                        regulation of metabolic process  BP 4221 404
## GO:0005488                                                binding  MF 7993 876
## GO:0032502                                  developmental process  BP 3334 319
## GO:0031323               regulation of cellular metabolic process  BP 3857 361
## GO:0048731                                     system development  BP 2484 228
## GO:0007275                     multicellular organism development  BP 2804 261
## GO:0016043                        cellular component organization  BP 3824 409
## GO:0006996                                 organelle organization  BP 2732 266
## GO:0010604 positive regulation of macromolecule metabolic process  BP 2214 224
## GO:0051171      regulation of nitrogen compound metabolic process  BP 3590 321
## GO:0080090                regulation of primary metabolic process  BP 3689 332
##            Down P.Up  P.Down
## GO:0050794 1177    1 5.2e-68
## GO:0050789 1206    1 6.1e-66
## GO:0065007 1233    1 6.5e-61
## GO:0048522  746    1 4.5e-50
## GO:0005515 1069    1 6.7e-50
## GO:0048518  779    1 1.6e-47
## GO:0060255  808    1 7.0e-46
## GO:0032501  754    1 1.0e-45
## GO:0048856  673    1 1.3e-45
## GO:0019222  853    1 1.4e-45
## GO:0005488 1360    1 1.5e-45
## GO:0032502  716    1 2.1e-45
## GO:0031323  796    1 6.2e-45
## GO:0048731  575    1 1.4e-44
## GO:0007275  628    1 1.7e-44
## GO:0016043  789    1 2.7e-44
## GO:0006996  607    1 2.5e-41
## GO:0010604  518    1 1.4e-40
## GO:0051171  738    1 1.8e-39
## GO:0080090  753    1 2.2e-39</code></pre>
<p>A simple way to visualise the enrichment results is through a bar plot as shown below which shows the top 10 enriched terms in each direction.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">top.up &lt;-<span class="st"> </span><span class="kw">head</span>(go.rst[<span class="kw">order</span>(go.rst<span class="op">$</span>P.Up, <span class="dt">decreasing =</span> F),], <span class="dt">n =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">top.down &lt;-</a>
<a class="sourceLine" id="cb41-3" data-line-number="3"><span class="st">  </span><span class="kw">head</span>(go.rst[<span class="kw">order</span>(go.rst<span class="op">$</span>P.Down, <span class="dt">decreasing =</span> F),], <span class="dt">n =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb41-4" data-line-number="4">bar.data &lt;-</a>
<a class="sourceLine" id="cb41-5" data-line-number="5"><span class="st">  </span><span class="kw">rbind</span>(</a>
<a class="sourceLine" id="cb41-6" data-line-number="6">    <span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb41-7" data-line-number="7">      <span class="dt">Term =</span> top.up<span class="op">$</span>Term,</a>
<a class="sourceLine" id="cb41-8" data-line-number="8">      <span class="dt">P.value =</span> top.up<span class="op">$</span>P.Up,</a>
<a class="sourceLine" id="cb41-9" data-line-number="9">      <span class="dt">Dir =</span> <span class="st">&quot;up&quot;</span></a>
<a class="sourceLine" id="cb41-10" data-line-number="10">    ),</a>
<a class="sourceLine" id="cb41-11" data-line-number="11">    <span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb41-12" data-line-number="12">      <span class="dt">Term =</span> top.down<span class="op">$</span>Term,</a>
<a class="sourceLine" id="cb41-13" data-line-number="13">      <span class="dt">P.value =</span> top.down<span class="op">$</span>P.Down,</a>
<a class="sourceLine" id="cb41-14" data-line-number="14">      <span class="dt">Dir =</span> <span class="st">&quot;down&quot;</span></a>
<a class="sourceLine" id="cb41-15" data-line-number="15">    )</a>
<a class="sourceLine" id="cb41-16" data-line-number="16">  )</a>
<a class="sourceLine" id="cb41-17" data-line-number="17">bar.data &lt;-<span class="st"> </span>bar.data[<span class="kw">order</span>(bar.data<span class="op">$</span>P.value, <span class="dt">decreasing =</span> F),]</a>
<a class="sourceLine" id="cb41-18" data-line-number="18"><span class="kw">par</span>(<span class="dt">mai =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="dv">2</span>, <span class="fl">0.5</span>, <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb41-19" data-line-number="19">bb &lt;-<span class="st"> </span><span class="kw">barplot</span>(</a>
<a class="sourceLine" id="cb41-20" data-line-number="20">  <span class="op">-</span><span class="kw">log10</span>(bar.data<span class="op">$</span>P.value),</a>
<a class="sourceLine" id="cb41-21" data-line-number="21">  <span class="dt">horiz =</span> T,</a>
<a class="sourceLine" id="cb41-22" data-line-number="22">  <span class="dt">xlab =</span> <span class="st">&quot; -log10(p-value)&quot;</span>,</a>
<a class="sourceLine" id="cb41-23" data-line-number="23">  <span class="dt">cex.names =</span> <span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb41-24" data-line-number="24">  <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)[<span class="kw">factor</span>(bar.data<span class="op">$</span>Dir)]</a>
<a class="sourceLine" id="cb41-25" data-line-number="25">)</a>
<a class="sourceLine" id="cb41-26" data-line-number="26"><span class="kw">axis</span>(</a>
<a class="sourceLine" id="cb41-27" data-line-number="27">  <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb41-28" data-line-number="28">  <span class="dt">line =</span> <span class="fl">-0.8</span>,</a>
<a class="sourceLine" id="cb41-29" data-line-number="29">  <span class="dt">at =</span> bb,</a>
<a class="sourceLine" id="cb41-30" data-line-number="30">  <span class="dt">labels =</span> bar.data<span class="op">$</span>Term,</a>
<a class="sourceLine" id="cb41-31" data-line-number="31">  <span class="dt">tick =</span> F,</a>
<a class="sourceLine" id="cb41-32" data-line-number="32">  <span class="dt">las =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb41-33" data-line-number="33">  <span class="dt">cex.axis =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb41-34" data-line-number="34">)</a>
<a class="sourceLine" id="cb41-35" data-line-number="35"><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>,<span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;Up&quot;</span>,<span class="st">&quot;Down&quot;</span>),<span class="dt">pch =</span> <span class="dv">22</span>,<span class="dt">pt.bg =</span> <span class="kw">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:enrichplot"></span>
<img src="bookdown-RNA-seq-protocol_files/figure-html/enrichplot-1.png" alt="Enrichment plot. Top enrichments in either direction" width="672" />
<p class="caption">
Figure 6.4: Enrichment plot. Top enrichments in either direction
</p>
</div>
<p><strong>Add gene information (optional)</strong></p>
<p>Sometimes you may want to know what genes are enriched in each of the terms, this section allows you to add gene details.</p>
<p>First, we extract the genes that are associated with the catalytic activity GO terms. This results in a list mapping the GO term and the associated genes.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">library</span>(GO.db)</a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="kw">library</span>(org.Mm.eg.db)</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">go.EntrezID &lt;-<span class="st"> </span><span class="kw">as.list</span>(org.Mm.egGO2ALLEGS)</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">go.rst &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(go.rst, <span class="dt">var =</span> <span class="st">&quot;GO.ID&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="co">#Add gene details</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2">go.rst<span class="op">$</span>Genes.Up &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(go.rst<span class="op">$</span>GO.ID, <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb43-3" data-line-number="3">  xx &lt;-<span class="st"> </span>go.EntrezID[[x]]</a>
<a class="sourceLine" id="cb43-4" data-line-number="4">  <span class="cf">if</span> (<span class="kw">is.null</span>(xx) <span class="op">|</span></a>
<a class="sourceLine" id="cb43-5" data-line-number="5"><span class="st">      </span><span class="kw">with</span>(go.rst[go.rst<span class="op">$</span>GO.ID <span class="op">==</span><span class="st"> </span>x,], Up <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">|</span></a>
<a class="sourceLine" id="cb43-6" data-line-number="6"><span class="st">           </span>P.Up <span class="op">&gt;</span><span class="st"> </span>enrich.pvalue))</a>
<a class="sourceLine" id="cb43-7" data-line-number="7">    <span class="kw">return</span>(<span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb43-8" data-line-number="8">  x &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">with</span>(de.genes,EntrezID[logFC<span class="op">&gt;</span><span class="dv">0</span>]),xx)</a>
<a class="sourceLine" id="cb43-9" data-line-number="9">  x &lt;-</a>
<a class="sourceLine" id="cb43-10" data-line-number="10"><span class="st">    </span><span class="kw">paste0</span>(<span class="kw">with</span>(fit.contr<span class="op">$</span>genes, Symbol[EntrezID <span class="op">%in%</span><span class="st"> </span>xx]), <span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>)</a>
<a class="sourceLine" id="cb43-11" data-line-number="11">  <span class="kw">return</span>(x)</a>
<a class="sourceLine" id="cb43-12" data-line-number="12">}))</a>
<a class="sourceLine" id="cb43-13" data-line-number="13">go.rst<span class="op">$</span>Genes.Down &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(go.rst<span class="op">$</span>GO.ID, <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb43-14" data-line-number="14">  xx &lt;-<span class="st"> </span>go.EntrezID[[x]]</a>
<a class="sourceLine" id="cb43-15" data-line-number="15">  <span class="cf">if</span> (<span class="kw">is.null</span>(xx) <span class="op">|</span></a>
<a class="sourceLine" id="cb43-16" data-line-number="16"><span class="st">      </span><span class="kw">with</span>(go.rst[go.rst<span class="op">$</span>GO.ID <span class="op">==</span><span class="st"> </span>x,], Down <span class="op">==</span><span class="st"> </span><span class="dv">0</span> <span class="op">|</span></a>
<a class="sourceLine" id="cb43-17" data-line-number="17"><span class="st">           </span>P.Down <span class="op">&gt;</span><span class="st"> </span>enrich.pvalue))</a>
<a class="sourceLine" id="cb43-18" data-line-number="18">    <span class="kw">return</span>(<span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb43-19" data-line-number="19">  x &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">with</span>(de.genes,EntrezID[logFC<span class="op">&lt;</span><span class="dv">0</span>]),xx)</a>
<a class="sourceLine" id="cb43-20" data-line-number="20">  x &lt;-</a>
<a class="sourceLine" id="cb43-21" data-line-number="21"><span class="st">    </span><span class="kw">paste0</span>(<span class="kw">with</span>(fit.contr<span class="op">$</span>genes, Symbol[EntrezID <span class="op">%in%</span><span class="st"> </span>xx]), <span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>)</a>
<a class="sourceLine" id="cb43-22" data-line-number="22">  <span class="kw">return</span>(x)</a>
<a class="sourceLine" id="cb43-23" data-line-number="23">}))</a></code></pre></div>
</div>
<div id="kegg-pathway-enrichment" class="section level3">
<h3><span class="header-section-number">6.3.2</span> KEGG pathway enrichment</h3>
<p>Pathway enrichment can also be performed against the Kyoto Encyclopedia of Genes and Genomes (KEGG) (<a href="https://www.genome.jp/kegg/" class="uri">https://www.genome.jp/kegg/</a>) database. The KEGG database contains curated molecular pathways and disease signatures. To perform this analysis we use the <em>kegga</em> function within the <strong>limma</strong> which is similar to the <em>goana</em> function we used above.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">kegg.rst&lt;-<span class="kw">kegga</span>(fit.contr,<span class="dt">species=</span><span class="st">&quot;Mm&quot;</span>)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="kw">topKEGG</span>(kegg.rst)</a></code></pre></div>
<pre><code>##                                               Pathway    N  Up Down    P.Up
## path:mmu03008       Ribosome biogenesis in eukaryotes   75  37    4 1.2e-14
## path:mmu01100                      Metabolic pathways 1014 200   75 2.2e-11
## path:mmu04810        Regulation of actin cytoskeleton  143   6   45 1.0e+00
## path:mmu05206                     MicroRNAs in cancer  121   6   40 1.0e+00
## path:mmu04919       Thyroid hormone signaling pathway   93   3   32 1.0e+00
## path:mmu03010                                Ribosome  137  39    1 6.1e-07
## path:mmu03020                          RNA polymerase   29  14    1 3.2e-06
## path:mmu01240               Biosynthesis of cofactors  112  32    2 5.6e-06
## path:mmu04330                 Notch signaling pathway   43   2   18 9.8e-01
## path:mmu00310                      Lysine degradation   52   1   20 1.0e+00
## path:mmu04611                     Platelet activation   76   0   25 1.0e+00
## path:mmu04062             Chemokine signaling pathway  119   5   34 1.0e+00
## path:mmu00190               Oxidative phosphorylation  117  31    0 4.0e-05
## path:mmu05200                      Pathways in cancer  323  26   71 1.0e+00
## path:mmu05205                 Proteoglycans in cancer  126   8   34 9.9e-01
## path:mmu04658        Th1 and Th2 cell differentiation   69   6   22 8.9e-01
## path:mmu05202 Transcriptional misregulation in cancer  120  16   32 4.6e-01
## path:mmu05224                           Breast cancer   81   8   24 8.2e-01
## path:mmu03050                              Proteasome   44  15    0 2.1e-04
## path:mmu05012                       Parkinson disease  206  44   12 3.1e-04
##                P.Down
## path:mmu03008 1.0e+00
## path:mmu01100 1.0e+00
## path:mmu04810 4.7e-08
## path:mmu05206 5.7e-08
## path:mmu04919 4.3e-07
## path:mmu03010 1.0e+00
## path:mmu03020 9.9e-01
## path:mmu01240 1.0e+00
## path:mmu04330 6.4e-06
## path:mmu00310 9.3e-06
## path:mmu04611 1.9e-05
## path:mmu04062 2.1e-05
## path:mmu00190 1.0e+00
## path:mmu05200 4.1e-05
## path:mmu05205 7.5e-05
## path:mmu04658 1.0e-04
## path:mmu05202 1.5e-04
## path:mmu05224 1.8e-04
## path:mmu03050 1.0e+00
## path:mmu05012 1.0e+00</code></pre>
</div>
<div id="gene-set-enrichment-analysis" class="section level3">
<h3><span class="header-section-number">6.3.3</span> Gene Set Enrichment Analysis</h3>
<p>GO and KEGG enrichment analysis uses the list of differentially expressed genes to test for over-representation of these genes. As such the results typically depend on the cut-offs used to get the differentially expressed genes.</p>
<p>Gene Set Enrichment Analysis is instead performed using <em>fry</em>, a gene set enrichment function within the <strong>limma</strong> package. Unlike above where we focused on a subset of genes(differentially genes), here all the genes in the set are used to test for gene expression signatures or pathways. A test is performed against the GO terms for demonstrative purposes, alternatively if you wish to test against a specific gene set, this can easily be done by importing that list.</p>
<p>We then use <em>fry</em> to test for enrichment between CIS and WT. We use the design matrix ‘design’,the expression values ‘y’ and the contrast matrix ‘contr’ from the imported results to perform the gene set test and the GO term and gene map that we created above.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">go.all&lt;-go.EntrezID[go.rst<span class="op">$</span>GO.ID]</a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="kw">names</span>(go.all)&lt;-go.rst<span class="op">$</span>Term</a>
<a class="sourceLine" id="cb46-3" data-line-number="3">fry.GSEA&lt;-<span class="kw">fry</span>(y,<span class="dt">index =</span> go.all,<span class="dt">design =</span> design,<span class="dt">contrast =</span> contr)</a>
<a class="sourceLine" id="cb46-4" data-line-number="4"><span class="kw">head</span>(fry.GSEA)</a></code></pre></div>
<pre><code>##                                                        NGenes Direction  PValue
## meiotic nuclear division                                  108      Down 2.3e-07
## chromosomal region                                        274      Down 3.2e-06
## chromosome organization involved in meiotic cell cycle     42      Down 1.3e-04
## meiotic chromosome segregation                             60      Down 1.6e-04
## chromosome, centromeric region                            187      Down 3.3e-04
## 90S preribosome                                            28        Up 5.1e-04
##                                                            FDR PValue.Mixed
## meiotic nuclear division                               0.00018       0.0131
## chromosomal region                                     0.00125       0.0162
## chromosome organization involved in meiotic cell cycle 0.03044       0.0094
## meiotic chromosome segregation                         0.03044       0.0079
## chromosome, centromeric region                         0.05207       0.0145
## 90S preribosome                                        0.05876       0.0039
##                                                        FDR.Mixed
## meiotic nuclear division                                   0.034
## chromosomal region                                         0.034
## chromosome organization involved in meiotic cell cycle     0.034
## meiotic chromosome segregation                             0.034
## chromosome, centromeric region                             0.034
## 90S preribosome                                            0.034</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">term.test&lt;-<span class="kw">rownames</span>(fry.GSEA)[<span class="dv">1</span>]</a></code></pre></div>
<p>Next, we can visualise the gene set enrichment results using a barcode plot. The barcode plot is used to plot the positions of genes within a set in ranked list of statistics. The statistics are ranked from left to right ie. from smallest to largest.</p>
<p>To do so, the gene identifies for the term should first be mapped to the index position of each gene in the expression matrix;</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">index&lt;-<span class="kw">rownames</span>(fit.contr) <span class="op">%in%</span><span class="st"> </span>go.all[[<span class="kw">rownames</span>(fry.GSEA)[<span class="dv">1</span>]]]</a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="kw">barcodeplot</span>(fit.contr<span class="op">$</span>coefficients[,<span class="st">&quot;CISvsWT&quot;</span>],<span class="dt">index =</span> index,<span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;CS&quot;</span>,<span class="st">&quot;WT&quot;</span>),</a>
<a class="sourceLine" id="cb49-3" data-line-number="3">            <span class="dt">main=</span><span class="kw">rownames</span>(fry.GSEA)[<span class="dv">1</span>])</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-34"></span>
<img src="bookdown-RNA-seq-protocol_files/figure-html/unnamed-chunk-34-1.png" alt="Barcode plot showing the enrichment of meiotic nuclear division in CS vs WT" width="672" />
<p class="caption">
Figure 6.5: Barcode plot showing the enrichment of meiotic nuclear division in CS vs WT
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="more-details-on-the-protocol.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-RNA-seq-protocol.pdf", "bookdown-RNA-seq-protocol.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
