<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Prerequisites | An RNA-seq Data Analysis Protocol</title>
  <meta name="description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Prerequisites | An RNA-seq Data Analysis Protocol" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Prerequisites | An RNA-seq Data Analysis Protocol" />
  
  <meta name="twitter:description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  

<meta name="author" content="Wei Shi" />


<meta name="date" content="2021-01-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="running-environment.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bulk RNA-seq analysis protocol</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a><ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>2.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>3</b> Running Environment</a><ul>
<li class="chapter" data-level="3.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>3.1</b> RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="running-environment.html"><a href="running-environment.html#unix-server-laptop"><i class="fa fa-check"></i><b>3.2</b> UNIX server + laptop</a></li>
<li class="chapter" data-level="3.3" data-path="running-environment.html"><a href="running-environment.html#laptop-only"><i class="fa fa-check"></i><b>3.3</b> Laptop only</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="rna-seq-analysis-protocol.html"><a href="rna-seq-analysis-protocol.html"><i class="fa fa-check"></i><b>4</b> RNA-SEQ analysis protocol</a></li>
<li class="chapter" data-level="5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html"><i class="fa fa-check"></i><b>5</b> More details on the protocol</a><ul>
<li class="chapter" data-level="5.1" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sequencing-quality"><i class="fa fa-check"></i><b>5.1</b> Sequencing quality</a></li>
<li class="chapter" data-level="5.2" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-annotation"><i class="fa fa-check"></i><b>5.2</b> Gene annotation</a></li>
<li class="chapter" data-level="5.3" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#index-building"><i class="fa fa-check"></i><b>5.3</b> Index building</a></li>
<li class="chapter" data-level="5.4" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-mapping"><i class="fa fa-check"></i><b>5.4</b> Read mapping</a></li>
<li class="chapter" data-level="5.5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-counting"><i class="fa fa-check"></i><b>5.5</b> Read counting</a></li>
<li class="chapter" data-level="5.6" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>5.6</b> Gene filtering</a></li>
<li class="chapter" data-level="5.7" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sample-clustering"><i class="fa fa-check"></i><b>5.7</b> Sample clustering</a></li>
<li class="chapter" data-level="5.8" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#statistical-testing-of-differential-expression"><i class="fa fa-check"></i><b>5.8</b> Statistical testing of differential expression</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An RNA-seq Data Analysis Protocol</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="prerequisites" class="section level1">
<h1><span class="header-section-number">2</span> Prerequisites</h1>
<p>Before you get started with the rest of the analysis, it is important that you have the necessary data and software that will be used in this analysis.</p>
<div id="data" class="section level2">
<h2><span class="header-section-number">2.1</span> Data</h2>
<p><strong>RNA-seq data</strong><br />
An RNA-seq dataset generated in a published study <span class="citation">(Delconte et al. <a href="#ref-delconte2016cis">2016</a>)</span> is used as an example dataset in this protocol.<br />
</p>
<!-- Two samples (wild-type and Cish-/- natural killer cells) are included in this dataset and each sample has two biological replicates. These data were already deposited in the Gene Expression Omnibus database (GSE79409). However for the convenience of this analysis,FASTQ files of the raw RNA-seq data were also saved in the ‘Workshop_RNAseq’ directory on Z drive (All Staff Share).  -->
<!-- A text file called "Targets.txt", which includes relevant sample information, can also be found in this directory. -->
<p><strong>Reference genome data</strong><br />
</p>
<!-- A FASTA-format file including all chromosomal sequences of the GRCm38/mm10 genome was also saved in ‘Workshop_RNAseq’ directory on Z drive. -->
</div>
<div id="software" class="section level2">
<h2><span class="header-section-number">2.2</span> SOFTWARE</h2>
<p>The following software tools should be installed on a UNIX server and on your laptop:</p>
<ul>
<li>R (<a href="https://www.r-project.org" class="uri">https://www.r-project.org</a>)</li>
<li>Rsubread (<a href="http://bioconductor.org/packages/release/bioc/html/Rsubread.html" class="uri">http://bioconductor.org/packages/release/bioc/html/Rsubread.html</a>)</li>
<li>limma (<a href="http://bioconductor.org/packages/release/bioc/html/limma.html" class="uri">http://bioconductor.org/packages/release/bioc/html/limma.html</a>)</li>
<li>edgeR (<a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html" class="uri">http://bioconductor.org/packages/release/bioc/html/edgeR.html</a>)</li>
<li>org.Mm.eg.db (<a href="http://bioconductor.org/packages/release/data/annotation/html/org.Mm.eg.db.html" class="uri">http://bioconductor.org/packages/release/data/annotation/html/org.Mm.eg.db.html</a>)</li>
<li>statmod (<a href="https://CRAN.R-project.org/package=statmod" class="uri">https://CRAN.R-project.org/package=statmod</a>)</li>
</ul>
<p>Consult the R Project website for the installation of R (<a href="https://www.r-project.org/" class="uri">https://www.r-project.org/</a>) <span class="citation">(R Core Team <a href="#ref-R-base">2020</a>)</span>. Make sure the latest release version of R is downloaded and installed. After R is installed, launch R and type the following commands to install <em>Rsubread</em>, <em>limma</em>, <em>edgeR</em>, <em>org.Mm.eg.db</em> and <em>statmod</em>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;Rsubread&quot;</span>,<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;edgeR&quot;</span>,<span class="st">&quot;org.Mm.eg.db&quot;</span>))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">install.packages</span>(<span class="st">&quot;statmod&quot;</span>)</a></code></pre></div>
<p><strong>Alternatively, you may use <a href="https://rstudio.com/">Rstudio</a> to run this protocol.</strong></p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-delconte2016cis">
<p>Delconte, Rebecca B, Tatiana B Kolesnik, Laura F Dagley, Jai Rautela, Wei Shi, Eva M Putz, Kimberley Stannard, et al. 2016. “CIS Is a Potent Checkpoint in Nk Cell–Mediated Tumor Immunity.” <em>Nature Immunology</em> 17 (7). Nature Publishing Group: 816–24.</p>
</div>
<div id="ref-R-base">
<p>R Core Team. 2020. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="running-environment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-RNA-seq-protocol.pdf", "bookdown-RNA-seq-protocol.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
