<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 RNA-SEQ analysis protocol | An RNA-seq Data Analysis Protocol</title>
  <meta name="description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 RNA-SEQ analysis protocol | An RNA-seq Data Analysis Protocol" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 RNA-SEQ analysis protocol | An RNA-seq Data Analysis Protocol" />
  
  <meta name="twitter:description" content="This is a protocol for running a Bulk RNA-seq analysis from raw-data to counts" />
  

<meta name="author" content="Wei Shi" />


<meta name="date" content="2021-01-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="running-environment.html"/>
<link rel="next" href="more-details-on-the-protocol.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bulk RNA-seq analysis protocol</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a><ul>
<li class="chapter" data-level="2.1" data-path="prerequisites.html"><a href="prerequisites.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>2.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>3</b> Running Environment</a><ul>
<li class="chapter" data-level="3.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>3.1</b> RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="running-environment.html"><a href="running-environment.html#unix-server-laptop"><i class="fa fa-check"></i><b>3.2</b> UNIX server + laptop</a></li>
<li class="chapter" data-level="3.3" data-path="running-environment.html"><a href="running-environment.html#laptop-only"><i class="fa fa-check"></i><b>3.3</b> Laptop only</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="rna-seq-analysis-protocol.html"><a href="rna-seq-analysis-protocol.html"><i class="fa fa-check"></i><b>4</b> RNA-SEQ analysis protocol</a></li>
<li class="chapter" data-level="5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html"><i class="fa fa-check"></i><b>5</b> More details on the protocol</a><ul>
<li class="chapter" data-level="5.1" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sequencing-quality"><i class="fa fa-check"></i><b>5.1</b> Sequencing quality</a></li>
<li class="chapter" data-level="5.2" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-annotation"><i class="fa fa-check"></i><b>5.2</b> Gene annotation</a></li>
<li class="chapter" data-level="5.3" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#index-building"><i class="fa fa-check"></i><b>5.3</b> Index building</a></li>
<li class="chapter" data-level="5.4" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-mapping"><i class="fa fa-check"></i><b>5.4</b> Read mapping</a></li>
<li class="chapter" data-level="5.5" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#read-counting"><i class="fa fa-check"></i><b>5.5</b> Read counting</a></li>
<li class="chapter" data-level="5.6" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>5.6</b> Gene filtering</a></li>
<li class="chapter" data-level="5.7" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#sample-clustering"><i class="fa fa-check"></i><b>5.7</b> Sample clustering</a></li>
<li class="chapter" data-level="5.8" data-path="more-details-on-the-protocol.html"><a href="more-details-on-the-protocol.html#statistical-testing-of-differential-expression"><i class="fa fa-check"></i><b>5.8</b> Statistical testing of differential expression</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An RNA-seq Data Analysis Protocol</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-analysis-protocol" class="section level1">
<h1><span class="header-section-number">4</span> RNA-SEQ analysis protocol</h1>
<p><em>Please note that your analysis results might be slightly different from those shown in this protocol due to software version differences.</em></p>
<p>We first change the working directory to the Workshop folder</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">setwd</span>(<span class="st">&quot;Workshop_RNAseq/&quot;</span>)</a></code></pre></div>
<p><strong>Build index for a reference genome</strong></p>
<p><strong>Step 1</strong><br />
Start an R session and build an index for the reference genome <em>GRCm38/mm10</em>. The created index files will be saved to the current working directory. This index only needs to be built once and it can be reused in future RNA-seq data analyses. <strong>TIMING ~40 mins</strong></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(Rsubread)</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">buildindex</span>(<span class="st">&quot;mm10_reference&quot;</span>, <span class="st">&quot;mm10.fa&quot;</span>)</a></code></pre></div>
<p>If you build an index on a laptop, it is recommended to build a split index. The amount of requested memory should be roughly half of the total memory available on your laptop. For example, if your laptop has 8GB memory, you may use the following command to build a split index with 4GB memory usage. <strong>TIMING ~90 mins</strong></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">buildindex</span>(<span class="st">&quot;mm10_reference&quot;</span>,</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">           <span class="st">&quot;mm10.fa&quot;</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">           <span class="dt">indexSplit =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">           <span class="dt">memory =</span> <span class="dv">4000</span>)</a></code></pre></div>
<p><strong>Create sample-related information and evaluate quality of sequencing TIMING ~ 3 mins</strong></p>
<p><strong>Step 2</strong><br />
Create a tab-delimited text file that contains sample-related information such as FASTQ file names, sample names and cell types. For convenience, a file called “Targets.txt” has already been created for this dataset and we read in this file:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">targets &lt;-<span class="st"> </span><span class="kw">readTargets</span>(<span class="st">&quot;Targets.txt&quot;</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">targets</a></code></pre></div>
<pre><code>##                                   Library   Sample CellType
## 1 CIS_1_C6AG6ANXX_ATCACG_L008_R1.fastq.gz CIS_rep1      CIS
## 2 CIS_2_C6AG6ANXX_CGATGT_L008_R1.fastq.gz CIS_rep2      CIS
## 3  WT_3_C6AG6ANXX_TTAGGC_L008_R1.fastq.gz  WT_rep1       WT
## 4  WT_4_C6AG6ANXX_TGACCA_L008_R1.fastq.gz  WT_rep2       WT</code></pre>
<p><strong>Step 3</strong><br />
Choose one of the libraries and examine the Phred quality scores of read bases at each read position. Boxplots of Phred scores are generated for each base position.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">qs &lt;-<span class="st"> </span><span class="kw">qualityScores</span>(targets<span class="op">$</span>Library[<span class="dv">1</span>], <span class="dt">offset =</span> <span class="dv">33</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">boxplot</span>(qs,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">        <span class="dt">xlab =</span> <span class="st">&quot;Base position&quot;</span>,</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">        <span class="dt">ylab =</span> <span class="st">&quot;Phred score&quot;</span>,</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">        <span class="dt">cex =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<p><strong>Align and count reads TIMING ~6 min per library</strong></p>
<p><strong>Step 4</strong><br />
Map sequence reads to mouse genome GRCm38/mm10. It is recommended to provide gene annotation to the read mapping. Mapped reads are saved to BAM files and a mapping summary is returned to R after read mapping is completed.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">align.outfiles &lt;-<span class="st"> </span><span class="kw">paste</span>(targets<span class="op">$</span>Sample, <span class="st">&quot;bam&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;.&quot;</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">align.summary &lt;-</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">  </span><span class="kw">align</span>(</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">    <span class="st">&quot;mm10_reference&quot;</span>,</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">    targets<span class="op">$</span>Library,</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">    <span class="dt">output_file =</span> align.outfiles,</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">    <span class="dt">nthreads =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">    <span class="dt">useAnnotation =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">    <span class="dt">annot.inbuilt =</span> <span class="st">&quot;mm10&quot;</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10">  )</a>
<a class="sourceLine" id="cb9-11" data-line-number="11">align.summary</a></code></pre></div>
<p><strong>Step 5</strong><br />
Assign reads to overlapping genes and generate an R object that contains a count table, gene annotation and counting summary. The count table contains number of reads assigned to each gene in each library. The gene annotation includes Entrez gene identifier, chromosomal coordinates of gene exons and gene length (total number of non-overlapping exonic bases each gene has). The counting summary gives the number of successfully assigned reads in each library and also numbers of reads that failed to be assigned due to filtering. The ‘Status’ column in this summary includes a ‘Assigned’ category and also multiple ‘Unassigned’ categories corresponding to different read filters used in counting.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">counts.gene &lt;-<span class="st"> </span><span class="kw">featureCounts</span>(align.outfiles, <span class="st">&quot;mm10&quot;</span>, <span class="dt">nthreads =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">counts.gene<span class="op">$</span>stat</a></code></pre></div>
<p><strong>Save R data and switch from the UNIX server to a personal computer TIMING &lt; 1 min</strong></p>
<p><strong>(Steps 6 and 7 can be skipped if you run the protocol using ‘RStudio (2672)’ or laptop only)</strong></p>
<p><strong>Step 6 (optional)</strong><br />
</p>
<p>Save all generated R objects to a file and then copy the file to your personal computer.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">save.image</span>(<span class="st">&quot;Counts.rdata&quot;</span>)</a></code></pre></div>
<p><strong>Step 7 (optional)</strong><br />
</p>
<p>Launch R on your personal computer and load R objects from the copied file.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;Counts.rdata&quot;</span>)</a></code></pre></div>
<p><strong>Perform differential expression analysis TIMING &lt; 1 min</strong></p>
<p><strong>Step 8</strong><br />
</p>
<p>Load required libraries.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">library</span>(statmod)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(org.Mm.eg.db))</a></code></pre></div>
<p><strong>Step 9</strong><br />
</p>
<p>Create a design matrix.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">ct &lt;-<span class="st"> </span><span class="kw">factor</span>(targets<span class="op">$</span>CellType)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>ct)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">levels</span>(ct)</a></code></pre></div>
<p><strong>Step 10</strong><br />
</p>
<p>Map Entrez gene identifiers to gene symbols and create an R object containing annotation data for each gene. The annotation data include Entrez gene identifier, gene symbol and gene length.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">tmp &lt;-<span class="st"> </span>org.Mm.egSYMBOL</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">entrez_symbol &lt;-<span class="st"> </span><span class="kw">as.list</span>(tmp[<span class="kw">mappedkeys</span>(tmp)])</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">entrez_symbol &lt;-<span class="st"> </span><span class="kw">sapply</span>(entrez_symbol, <span class="cf">function</span>(x)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">  x[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">genes &lt;-<span class="st"> </span>counts.gene<span class="op">$</span>annotation<span class="op">$</span>GeneID</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">m &lt;-<span class="st"> </span><span class="kw">match</span>(genes, <span class="kw">names</span>(entrez_symbol))</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">genes &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">  <span class="dt">EntrezID =</span> genes,</a>
<a class="sourceLine" id="cb15-9" data-line-number="9">  <span class="dt">Symbol =</span> entrez_symbol[m],</a>
<a class="sourceLine" id="cb15-10" data-line-number="10">  <span class="dt">Length =</span> counts.gene<span class="op">$</span>annotation<span class="op">$</span>Length,</a>
<a class="sourceLine" id="cb15-11" data-line-number="11">  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12">)</a></code></pre></div>
<p><strong>Step 11</strong><br />
</p>
<p>Remove from analysis those genes that did not express, or expressed at a very low level, in all cell types. Here we require a gene to have a CPM value greater than 0.5 in at least two libraries to be included in the subsequent analysis. The reason we require at least two libraries is because there are two biological replicates generated for each sample in this dataset.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">cpm</span>(counts.gene<span class="op">$</span>counts) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw">sum</span>(keep)</a></code></pre></div>
<pre><code>## [1] 11969</code></pre>
<p><strong>Step 12</strong><br />
Transform count data to log2-cpm, estimate the mean-variance relationship and compute observation-level weights. Log2-cpm expression values of genes are then normalized using the quantile method and converted to log2-rpkm (log2 reads per kilobases of exons per million reads) values.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">y &lt;-</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="st">  </span><span class="kw">voom</span>(counts.gene<span class="op">$</span>counts[keep, ], design, <span class="dt">normalize.method =</span> <span class="st">&quot;quantile&quot;</span>)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">y<span class="op">$</span>genes &lt;-<span class="st"> </span>genes[keep, ]</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">y<span class="op">$</span>E &lt;-<span class="st"> </span>y<span class="op">$</span>E <span class="op">-</span><span class="st"> </span><span class="kw">log2</span>(y<span class="op">$</span>genes<span class="op">$</span>Length <span class="op">/</span><span class="st"> </span><span class="dv">1000</span>)</a></code></pre></div>
<p><strong>Step 13</strong><br />
Cluster samples via multidimensional scaling.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">plotMDS</span>(y, <span class="dt">labels=</span>targets<span class="op">$</span>CellType)</a></code></pre></div>
<p><img src="bookdown-RNA-seq-protocol_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p><strong>Step 14</strong><br />
</p>
<p>Fit linear models to genes and perform statistical testing to discover differentially expressed genes. A contrast matrix is constructed to specify the comparisons between samples. The empirical Bayes moderated t-statistic is used to assess the differential expression of genes. A FDR (false discovery rate) cutoff is applied for calling differentially expressed genes. For this analysis, an FDR cutoff of 0.05 was applied (default cutoff value in decideTests()).</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(y, design)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">contr &lt;-<span class="st"> </span><span class="kw">makeContrasts</span>(<span class="dt">CISvsWT =</span> CIS <span class="op">-</span><span class="st"> </span>WT, <span class="dt">levels =</span> design)</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">fit.contr &lt;-<span class="st"> </span><span class="kw">eBayes</span>(<span class="kw">contrasts.fit</span>(fit, contr))</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">dt &lt;-<span class="st"> </span><span class="kw">decideTests</span>(fit.contr)</a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="kw">summary</span>(dt)</a></code></pre></div>
<pre><code>##        CISvsWT
## Down      1662
## NotSig    8788
## Up        1519</code></pre>
<p><strong>Step 15</strong><br />
</p>
<p>Display top 10 most differentially expressed genes:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw">options</span>(<span class="dt">digits=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">topTable</span>(fit.contr)</a></code></pre></div>
<pre><code>##        EntrezID           Symbol Length logFC AveExpr   t P.Value adj.P.Val
## 12700     12700             Cish   2161  -3.5   4.705 -27 2.1e-08   7.4e-05
## 69368     69368            Wdfy1   4714  -2.6   1.999 -21 1.6e-07   3.1e-04
## 21638     21638            Trgv4    419   3.8   3.666  17 5.9e-07   8.8e-04
## 436468   436468 Trav15d-1-dv6d-1    349   4.2   3.098  17 4.8e-07   8.1e-04
## 22290     22290              Uty   5228 -11.0  -2.841 -31 9.7e-09   7.4e-05
## 26908     26908          Eif2s3y   1767 -12.2  -0.086 -27 2.4e-08   7.4e-05
## 26900     26900            Ddx3y   4640 -11.5  -1.825 -27 2.5e-08   7.4e-05
## 20592     20592            Kdm5d   5471 -10.6  -2.515 -25 4.0e-08   9.6e-05
## 59310     59310            Myl10    880   2.6   3.140  14 2.1e-06   2.3e-03
## 13508     13508            Dscam   7481  -2.3   0.022 -14 2.3e-06   2.3e-03
##           B
## 12700  10.1
## 69368   8.2
## 21638   6.6
## 436468  6.3
## 22290   6.1
## 26908   6.0
## 26900   6.0
## 20592   5.8
## 59310   5.7
## 13508   5.6</code></pre>
<p><strong>Step 16</strong><br />
</p>
<p>Save all differentially expressed genes to a file.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">de.genes &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit.contr, <span class="dt">coef=</span><span class="st">&quot;CISvsWT&quot;</span>, <span class="dt">p.value=</span><span class="fl">0.05</span>, <span class="dt">n=</span><span class="ot">Inf</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw">write.csv</span>(de.genes, <span class="dt">file=</span><span class="st">&quot;DE_genes.csv&quot;</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="running-environment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="more-details-on-the-protocol.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-RNA-seq-protocol.pdf", "bookdown-RNA-seq-protocol.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
